src/lib/decodeBySigHash.ts

```diff
@@ -25,7 +25,12 @@ export async function fetchAndDecodeWithCalldata(
   const data = await getSignaturesByCalldata(sigHash);
   if (data) {
     const ifaces = parse4BytesResToIfaces(data);
-    return decodeByCalldata(ifaces, calldata);
+    const decodedByCalldata = decodeByCalldata(ifaces, calldata);
+    if (decodedByCalldata.length === 0 && ifaces.length > 0) {
+      return [{ decoded: [], fragment: ifaces[0].fragments[0], sigHash }];
```

Comment:
Idk, in all of the cases if there are no results the decoded array is empty. It would be rather strange to change the whole type definition only for that one edge case. Especially, when there's nothing wrong IMO in signaling that there are no results when the array is empty. If the user did something wrong we still got errors that might explain if there was actually an error. If you still think that we should change it we can do that ofc.

Reply:
Ok. I forgot the context here. I thought those are parameters, not different decodings